src/dparse/parser.d

```diff
@@ -532,9 +532,20 @@ class Parser
             advance(); // align
             node.hasAlign = true;
             if (currentIsOneOf(tok!"intLiteral", tok!"identifier"))
+            {
                 node.identifierOrIntegerOrOpcode = advance();
+                if (!currentIs(tok!";"))
+                {
+                    error("';' expected.");
+                    advance();
+                    return null;
```

Comment:
The loop was `while (stack.put(<the result of this>){ ... }`. The temp node was always returned.